docs: adds oauth plain authentication flows (#6262)
strimzi/strimzi-kafka-operator: documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc

<5> (Only applicable when using `introspectionEndpointUri`) Depending on the authorization server you are using, the introspection endpoint may or may not return the _token type_ attribute, or it may contain different values. You can specify a valid token type value that the response from the introspection endpoint has to contain.
<6> (Only applicable when using `introspectionEndpointUri`) The authorization server may be configured or implemented in such a way to not provide any identifiable information in an Introspection Endpoint response. In order to obtain the user ID, you can configure the URI of the `userinfo` endpoint as a fallback. The `userNameClaim`, `fallbackUserNameClaim`, and `fallbackUserNamePrefix` settings are applied to the response of `userinfo` endpoint.
<7> Set this to `false` to disable the OAUTHBEARER mechanism on the listener. At least one of PLAIN or OAUTHBEARER has to be enabled. Default is `true`.
<8> Set to `true` to enable PLAIN authentication on the listener, which is supported for clients on all platforms.

> <9> Additional configuration for the PLAIN mechanism. Clients can authenticate by passing a client ID and secret as a `username` and `password`. Default is `false`. If not specified, clients can authenticate over PLAIN only by passing an access token as the `password` using an `$accessToken:` prefix.

Reviewer: I'm not sure what 9 refers to without seeing a rendered doc, but the only setting I can think of here is `tokenEndpointUri` which is not true / false.

```suggestion
<9> Additional configuration for the PLAIN mechanism. Clients can authenticate by passing a client ID and secret as a `username` and `password`, or by passing account name and access token with an `$accessToken:` prefix. If not specified, clients can authenticate over PLAIN only by passing an unprefixed access token as the `password`.
```

Author: Yes. It's the callout for the tokenEndpointUri. 
Maybe this is enough:

> Additional configuration for the PLAIN mechanism. If specified, clients can authenticate over PLAIN by passing an access token as a `password` using an `$accessToken:` prefix.

Reviewer: I don't know, maybe. I like to repeat myself in different places because usually we don't read docs from start to finish but go look for a solution. Here being reminded that how you pass access token depends on whether this is configured or not.

Author: Agreed. But this a callout. I don't think we need to be verbose. We have the options covered in a couple of other places.
I've added a link in the _Before you start_ pf this procedure that goes to the _Oauth 2.0 authentication mechanisms_ section as reference
